x/yarpctest/api/request_unary.go

```diff
@@ -24,17 +24,19 @@ import (
 	"bytes"
 	"time"
 
+	"go.uber.org/yarpc/api/middleware"
 	"go.uber.org/yarpc/api/transport"
 )
 
 // RequestOpts are configuration options for a yarpc Request and assertions
 // to make on the response.
 type RequestOpts struct {
-	Port         uint16
-	GiveTimeout  time.Duration
-	GiveRequest  *transport.Request
-	WantResponse *transport.Response
-	WantError    error
+	Port            uint16
+	UnaryMiddleware []middleware.UnaryOutbound
```

Comment:
My intuition is that middleware belong to ServiceOpts, not RequestOpts. Is there a rationale?

Reply:
The way the framework is structured, each created service is a long running inbound and each request creates an ad hoc transport outbound with a single peer.

The outbound middleware needs to be set on each request because the service does not have the opportunity to apply outbound middleware on the request, since the request has already been executed.